src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsUpdateStep.cs

```diff
@@ -0,0 +1,114 @@
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            // Project should be backed up before changing source
+            WellKnownStepIds.BackupStepId,
+
+            // Template files should be added prior to changing source (since some code fixers will change added templates)
+            WellKnownStepIds.TemplateInserterStepId,
+
+            // Project should have correct TFM
+            WellKnownStepIds.SetTFMStepId,
+        };
+
+        public override IEnumerable<string> DependencyOf { get; } = new[]
+        {
+            WellKnownStepIds.NextProjectStepId,
+        };
+
+        public WinformsUpdateStep(IEnumerable<IUpdater<IProject>> winformsUpdaters, ILogger<WinformsUpdateStep> logger)
+            : base(logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            SubSteps = new List<UpgradeStep>(winformsUpdaters.Select(updater => new WinformsUpdaterSubStep(this, updater, logger)));
+        }
+
+        /// <summary>
+        /// Determines whether the WinformsUpdaterStep applies to a given context.
+        /// </summary>
+        /// <param name="context">The context to evaluate.</param>
+        /// <param name="token">A token that can be used to cancel execution.</param>
+        /// <returns>True if the Winforms updater step might apply, false otherwise.</returns>
+        protected override async Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context?.CurrentProject is null)
+            {
+                return false;
+            }
+
+            foreach (var subStep in SubSteps)
+            {
+                if (await subStep.IsApplicableAsync(context, token).ConfigureAwait(false))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            foreach (var step in SubSteps)
+            {
+                await step.InitializeAsync(context, token).ConfigureAwait(false);
+            }
+
+            var incompleteSubSteps = SubSteps.Count(s => !s.IsDone);
+
+            return incompleteSubSteps == 0
+                ? new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "No Winforms updaters need applied", BuildBreakRisk.None)
+                : new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"{incompleteSubSteps} Winforms updaters need applied", SubSteps.Where(s => !s.IsDone).Max(s => s.Risk));
+        }
+
+        public override UpgradeStepInitializeResult Reset()
+        {
+            return base.Reset();
```

Comment:
this is unnecessary - it will do this by default. Do you plan on having custom logic?